cli/commands/run/server/agent.controller.spec.ts

```diff
@@ -387,5 +407,43 @@ describe("AgentController", () => {
       })
       expect(txEvent.contractAddress).toStrictEqual(formatAddress(mockTxRequest.request.event.contractAddress))
     })
+  
+    it("throws an error if more than 10 findings when handling a transaction", async () => {
+        const findings = (new Array(21)).fill(mockFinding)
+        mockHandleTransaction.mockReturnValue(findings)
+        mockGetAgentHandlers.mockReturnValue({ handleTransaction: mockHandleTransaction })
+        agentController = new AgentController(mockGetAgentHandlers)
+
+        await agentController.initializeAgentHandlers()
+        await agentController.EvaluateTx(mockTxRequest, mockCallback)
+
+        expect(mockCallback).toHaveBeenCalledWith(null, {
+          status: "ERROR",
+          findings: [],
+          metadata: {
+            timestamp: systemTime.toISOString(),
+          },
+          private: false
+        })
+    })
+
+    it("throws an error if more than 50kB of findings fetched when handling a transaction", async () => {
```

Comment:
should we add a similar test case for handling blocks?

Reply:
Added a similar test to handling a transaction. Nice catch!